metadata_backend/api/handlers.py

```diff
@@ -1133,23 +1134,72 @@ async def _extract_xml_upload(req: Request, extract_one: bool = False) -> List[T
         part = cast(BodyPartReader, part)
         if not part:
             break
-        if extract_one and files:
+        if extract_one and (xml_files or csv_files):
             reason = "Only one file can be sent to this endpoint at a time."
             LOG.error(reason)
             raise web.HTTPBadRequest(reason=reason)
-        if part.name:
-            schema_type = part.name.lower()
-            if schema_type not in schema_types:
-                reason = f"Specified schema {schema_type} was not found."
-                LOG.error(reason)
-                raise web.HTTPNotFound(reason=reason)
-            data = []
-            while True:
-                chunk = await part.read_chunk()
-                if not chunk:
-                    break
-                data.append(chunk)
-            xml_content = "".join(x.decode("UTF-8") for x in data)
-            files.append((xml_content, schema_type))
-            LOG.debug(f"processed file in {schema_type}")
-    return sorted(files, key=lambda x: schema_types[x[1]]["priority"])
+
+        content, schema_type = await _extract_upload(part)
+
+        if is_xml:
+            # Validation is skipped here because validation endpoint
+            # does its own validating on a supposed XML
+            xml_files.append((content, schema_type))
+        else:
+            # Multipart can have XML or CSV files
+            # Default expectation is XML, so XML is tried first
+            schema = XMLSchemaLoader().get_schema(schema_type)
```

Comment:
to test if it is XML you migth be able to use: ` XMLResource` from `xmlschema` package as you can see in tests: https://github.com/sissaschool/xmlschema/blob/bb528a94d83f9fc937765d16e266506bcf127318/tests/test_resources.py#L492-L494 `ParseError` is triggered when xml is not valid.  That will avoid us needing to load a schema.

Also a `try` `except` might be better here where we catch exact error e.g. if we ` except csv.Error:` then we do in that except a `try` with xml if that gives `except xmlschema.etree.ParseError:` then we fail as unknown format

Reply:
Trying `XMLResource` seems to work well. I changed this to try/excepts but as for the order of testing, the key issue is that validating CSV is difficult when something like `<SAMPLE_SET>,</SAMPLE_SET>` is instantly identifiable to us humans as XML but also clearly passes as valid CSV. Hence, with the current way, XML would need to be validated first and CSV after that.